src/raps/actions/swap.ts

```diff
@@ -323,25 +342,34 @@ export const swap = async ({
         asset: {
           ...parameters.assetToBuy,
           network: ethereumUtils.getNetworkFromChainId(parameters.assetToBuy.chainId),
-          colors: parameters.assetToSell.colors as TokenColors,
+          colors: parameters.assetToBuy.colors as TokenColors,
+          price: nativePriceForAssetToBuy,
         },
-        value: quote.buyAmount.toString(),
+        value: quote.buyAmountMinusFees.toString(),
       },
     ],
+    gasLimit,
     hash: swap.hash as TxHash,
     // TODO: MARK - Replace this once we migrate network => chainId
     network: ethereumUtils.getNetworkFromChainId(parameters.chainId),
     // chainId: parameters.chainId,
     nonce: swap.nonce,
     status: 'pending',
     type: 'swap',
+    swap: {
+      type: SwapType.normal,
+      fromChainId: parameters.assetToSell.chainId as unknown as SwapChainId,
+      toChainId: parameters.assetToBuy.chainId as unknown as SwapChainId,
+
+      // TODO: Is this right?
+      isBridge:
```

Comment:
what is this used for? it's also possible to execute a bridge tx with two different tokens, but not sure if that is relevant here

Reply:
I don't see it being used anywhere currently so not sure if it's even necessary. I can probably just remove it.